test/tftp/tftp_api_test.rb

```diff
@@ -26,22 +26,64 @@ def test_api_can_fetch_boot_file
   end
 
   def test_api_can_create_syslinux_tftp_reservation
```

Comment:
I think these tests should be broken into two or three suites:
 - a suite to test Proxy::TFTP::Server, in particular to verify that files are created (with content we expect), deleted, etc. I'd avoid mocking altogether for this, and rely on Tempfile instead (which are automatically deleted during garbage collection/on vm exit)
 - a suite of tests to verify paths for subclasses of ::Proxy::TFTP::Server
 - a suite of tests to verify tftp api (ok to mock Proxy::TFTP::Server interface)